### PR TITLE
Revamp 'available balance'

### PR DESCRIPTION
**Pull request by Patrick1904**
This PR serves as a prerequisite for [Wallet Onboarding: Coin-op 1.5](https://github.com/near/near-wallet/issues/2050). 

The purpose of this change is to simplify the 'available balance' displayed throughout the NEAR Wallet by removing any contextual 'available balance' that previously existed for sending and staking NEAR.

An amount 'reserved for transactions' (currently 0.1N) is now deducted from the users available balance. This is displayed in the balance breakdown on the `/profile` page.

Main changes includes:
1. Show only the **available balance** on the main dashboard, vs total balance (account + staking + lockup).
2. Staking and lockup-related balances can be found on the `/staking` and `/profile` pages.
3. Make the 'Wallet balance' shown on the `/profile` match the Explorer (Available balance + amount reserved for storage).
4. Remove any contextual deductions for the available balance while sending, staking, or signing transactions. The same 'available balance' now applies throughout the wallet.
5. Include the globally applied 'reserved for transactions' (0.1N) amount in the balance breakdown on the `/profile` to show how the available balance is calculated.
6. On the `/profile`, move out the staking balances to visually separate them from the main account balance.
7. Users with very small balances will now see a zero available balance since we're now deducting the amount 'reserved for transactions'. This means that any user with an available balance below 0.1N (amount could change) will see a zero available balance since anything below 0.1N is reserved. The deduction is currently set to _up to 0.1N_, meaning if the user only has 0.03N available, 0.03N will be reserved.
8. Replace `WALLET_APP_MIN_AMOUNT` with `MIN_BALANCE_FOR_GAS` globally.
9. Only show 'Staking from' when there are multiple accounts, i.e. main account + lockup account.

### CODE REVIEW

**Review by MaximusHaximus — CHANGES_REQUESTED** [review]
Looks awesome, just two things :+1:

File: packages/frontend/src/reducers/selectors/balance.js [hunk 1]
```diff
@@ -17,14 +25,18 @@ export const selectProfileBalance = (balance) => {
...
+        available
     } = balance;
 
     const lockupIdExists = !!lockedAmount;
 
     const walletBalance = {
-        walletBalance: stakedBalanceMainAccount.add(new BN(balanceAvailable)).add(new BN(stateStaked)).toString(),
+        walletBalance: walletAccount?.amount,
         reservedForStorage: stateStaked.toString(),
+        reservedForTransactions: new BN(available).sub(new BN(parseNearAmount(WALLET_APP_MIN_AMOUNT))).isNeg()
```
[the remark this answers is not in the document]
Comment (Patrick1904) [reply packages/frontend/src/reducers/selectors/balance.js:37]: Good point :)

**Review by corwinharrell — CHANGES_REQUESTED** [review]
Just a couple small things:

1. Let's make sure the styling of the staking section matches the one above (indentation + gray background)

Screen Shot 2021-09-01 at 6 53 26 PM

2. @MaximusHaximus made a good suggestion to keep the tooltip on the available balance, and to also include a text link that points to the account details page, just so it makes the transition a little easier. I think new can change the available balance text to:

> Your spendable NEAR balance excluding funds locked or staked. _View Balance Breakdown_ →

**Comment by corwinharrell** [comment]
Not at my computer for the rest of the night to review again so feel free to merge after those changes 👍

**Pull request merged**